ext/openssl/ossl_mac.c

```diff
@@ -0,0 +1,174 @@
+#include "ossl.h"
+
+#if OSSL_OPENSSL_PREREQ(3, 0, 0)
```

Comment:
I believe LibreSSL and AWS-LC will eventually implement the `EVP_MAC` API too. Please use an `extconf.rb` check instead of relying on the version number.